prow/cmd/plank/main.go

```diff
@@ -128,8 +131,20 @@ func main() {
 		logrus.WithError(err).Fatal("Error creating plank controller.")
 	}
 
+	pjClientset, err := o.kubernetes.ProwJobClientset(cfg().PodNamespace, o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create prowjob client set")
+	}
+	informerFactory := prowjobinformer.NewSharedInformerFactory(pjClientset, 15*time.Second)
```

Comment:
@alvaroaleman any negative effects here of starting this for the metrics collection? We could defer refactoring the rest of `plank` for later, right?

Reply:
This metric `prow_job_labels` is similiar to `kube_pod_labels` defined [pod-metrics](https://github.com/kubernetes/kube-state-metrics/blob/master/docs/pod-metrics.md).

The goal of this metric is to be used to do [joint query](https://github.com/kubernetes/kube-state-metrics/tree/master/docs#join-metrics), for example, alerting when a specific kind of jobs failed.

I do realize now what we are doing does sound like a prow-job exporter. With only 1 or 2 metrics defined, the exporter sound a bit thin for the moment.

I wonder which existing metrics can moved into exporter if we do exporter in the end.

Reply:
More specific on the use case:
`prow_job_labels` will expose all ever existing prow jobs with their labels.
The queries jointing with `prowjob_state_transitions` can be used for alerting on a label, e.g., `role: infra`. Without `prow_job_labels`, we could still do it with `prow_job_name` label and we have to do OR there for different names. Using more generic label, it is much easier.